src/BrickbattleWeapons/MainModule/ServerModules/slices/BrickbattleWeaponSlice.lua

```diff
@@ -0,0 +1,34 @@
+--!strict
+local Reflex = require("Packages/reflex")
+local Immut = require("Packages/immut")
+local Table = require("src/BrickbattleWeapons/MainModule/ServerModules/slices/BrickbattleWeaponTables")
+
+export type BrickbattleWeaponStateActions = {
+    setStatState: (playerName: string, numb: number, stat: string) -> (),
+    removeStatState: (playerName: string, stat: string) -> (),
+}
+
+type Data = {
+    [string]: Table.BrickbattleWeaponTable
+}
+local initialState: Data = {}
+local BrickbattleWeaponsStateSlice = Reflex.createProducer(initialState, {
```

Comment:
nitpick: since this slice is (apparently) specific to a player, let's clarify that in the name. I'd also drop "Weapon" table since eventually we'll need another table to keep track of reload times for a player's tools. let's just call it "BrickbattleStateSlice", unless you have a better idea.